src/components/PatternPlayground/PatternPlayground.tsx

```diff
@@ -99,15 +97,15 @@ export const PatternPlayground = observer(function PatternPlayground() {
         <PatternList
           selectedPatternBlock={selectedPatternBlock}
           onSelectPatternBlock={onSelectPatternBlock}
-          selectedEffectBlock={selectedEffectBlock}
+          selectedEffectIndices={selectedEffectIndices}
           onSelectEffectBlock={onSelectEffectBlock}
         />
       </GridItem>
       <GridItem area="controls">
         <ParameterControls block={selectedPatternBlock} />
-        {selectedEffectBlock && (
-          <ParameterControls block={selectedEffectBlock} />
-        )}
+        {selectedEffectIndices.map((effectIndex, i) => (
+            <ParameterControls block={effectBlocks[effectIndex]} />
```

Comment:
So VSCode alerted me that this needed a `key` prop since it is an element inside a loop. It looks like it's my ESLint extension letting me know in case you want to set that up in your editor!
```suggestion
            <ParameterControls key={i} block={effectBlocks[effectIndex]} />
```